ledger/ledger-api-test-tool/src/main/scala/com/daml/ledger/api/testtool/infrastructure/Assertions.scala

```diff
@@ -32,11 +36,58 @@ object Assertions extends DiffExtensions {
     val diff = Diff.compare(actual, expected)
     if (!diff.isIdentical)
       throw AssertionErrorWithPreformattedMessage(
-        diff.show,
+        diff.show(),
         s"$context: two objects are supposed to be equal but they are not",
       )
   }
 
+  // Specialize a few instances via semi-automatic derivation
```

Comment:
@kamil-da @mzagorski-da I had to make a bunch of changes to actually get this to work so please take another look if you’re fine with those changes:

1. Use `Diff.derived` instead of `implicitly`. Afaict, the default `implicitly` simply uses `==` so doesn’t produce any diffs which is clearly not the point here.
2. This then ran into a bunch of bugs in `diffx` which forced an upgrade to the latest version which I also included here.

Reply:
And now it fails on Scala 2.13, I hate implicits so much.

Reply:
The changes look good, thanks! Although it's a bit disappointing that we have to explicitly write all those definitions.